packages/components/codemods/renameV2ComponentImportsAndUsages/renameV2ComponentImportsAndUsages.spec.ts

```diff
@@ -0,0 +1,269 @@
+      })
+    }
+  </SingleSelect>
+);
+`)
+
+      const result = transformComponents(inputAst)
+      expect(result).toBe(printAst(expectedAst))
+    })
+
+    it('should not override local types of same name', () => {
+      const inputAst = parseJsx(`
+type SelectProps = {
+  thing: string
+}
+
+const MyComponent = (props: SelectProps) => {  
+  return <div>{props.customField}</div>
+}
+`)
+
+      const result = transformComponents(inputAst)
+      expect(result).toBe(printAst(inputAst))
+    })
+  })
+
+  describe('LikertScaleLegacy -> LikertScale', () => {
+    it('should rename LikertScaleLegacy import to LikertScale', () => {
+      const inputAst = parseJsx(`import { LikertScaleLegacy } from "@kaizen/components"`)
+      const expectedAst = parseJsx(`import { LikertScale } from "@kaizen/components"`)
+
+      const result = transformComponents(inputAst)
+      expect(result).toBe(printAst(expectedAst))
+    })
+
+    it('should rename LikertScaleLegacy JSX element to LikertScale', () => {
+      const inputAst = parseJsx(`
+import { LikertScaleLegacy } from "@kaizen/components"
+
+const MyComponent = () => <LikertScaleLegacy />
+`)
+      const expectedAst = parseJsx(`
+import { LikertScale } from "@kaizen/components"
+
+const MyComponent = () => <LikertScale />
+`)
+
+      const result = transformComponents(inputAst)
+      expect(result).toBe(printAst(expectedAst))
+    })
+
+    it('should leave LikertScaleProps type as is', () => {
+      const inputAst = parseJsx(`
+import type { LikertScaleProps } from "@kaizen/components"
+
+const MyComponent = (props: LikertScaleProps) => null
+`)
+      const result = transformComponents(inputAst)
+      expect(result).toBe(printAst(inputAst))
+    })
+  })
+
+  describe('TitleBlockZen -> TitleBlock', () => {
+    it('should rename TitleBlockZen import to TitleBlock', () => {
+      const inputAst = parseJsx(`import { TitleBlockZen } from "@kaizen/components"`)
+      const expectedAst = parseJsx(`import { TitleBlock } from "@kaizen/components"`)
+
+      const result = transformComponents(inputAst)
+      expect(result).toBe(printAst(expectedAst))
+    })
+
+    it('should rename TitleBlockZen JSX element to TitleBlock', () => {
+      const inputAst = parseJsx(`
+import { TitleBlockZen } from "@kaizen/components"
+
+const MyComponent = () => <TitleBlockZen />
+`)
+      const expectedAst = parseJsx(`
+import { TitleBlock } from "@kaizen/components"
+
+const MyComponent = () => <TitleBlock />
+`)
+
+      const result = transformComponents(inputAst)
+      expect(result).toBe(printAst(expectedAst))
+    })
+
+    it('should leave TitleBlockProps type as is', () => {
+      const inputAst = parseJsx(`
+import type { TitleBlockProps } from "@kaizen/components/next"
+
+const MyComponent = (props: TitleBlockProps) => null
+`)
+      const result = transformComponents(inputAst)
+      expect(result).toBe(printAst(inputAst))
+    })
+  })
+
+  describe('Mixed scenarios', () => {
```

Comment:
Oddly enough, while this works in our test enviroment, I came into a scenario where the TitleBlock is removed entirely when testing in the perf-review-cycle-ui in the manage app when running this in `/src`.

<img width="1247" height="177" alt="Screenshot 2025-07-28 at 10 59 57 am" src="https://github.com/user-attachments/assets/b13ff483-5b23-46d8-a2d9-92938d56c99f" />

Reply:
I lifted the exact code from `SubjectsSummaryLayout.tsx` and only thing I can find that might be causing the issue is that TitleBlockZen is used within a prop?

```JSX
const SubjectsSummary = ({ children }: PropsWithChildren) => {
 
  return (
    <DrawerLayout
      HeaderContent={
        <UnifiedNav>
          // fails
          <TitleBlockZen
            navigationTabs={props.map(prop => <NavigationTab {...prop} />)}
          />
        </UnifiedNav>
      }
    >
      // fine
      <TitleBlockZen
        navigationTabs={props.map(prop => <NavigationTab {...prop} />)}
      />
      {children}
    </DrawerLayout>
  )
}
)
`)
```